### ISSUE TITLE
Feature Request - Restrict Display to Bounds

### ISSUE THREAD

**Issue by tcoats** [Leaflet/Leaflet#211]
It would be useful to restrict the view of a given Leaflet to a specific area.

This could be used:
- In contact forms
- Where the area is important, for example a city map or a map of a specific area

I would expect Leaflet to never show map outside the restricted area, this includes:
- When zooming out, restricting an area would impose a minimum zoom
- Ideally this minimum zoom to be fractional - eg a zoom level between 4 and 5 displays the area bets
- When resizing, or automatically resizing the imposed minimum zoom would have to be updated

I'm considering writing this myself, any guidance would be most welcome.

In Summary:
- Map Options allowing a developer to restrict the area allowed to display in Leaflet
